### PR TITLE
Update frontend API contract

### PR DESCRIPTION
**Pull request by codemonkey800**
## Description

- Updates typing and hooks for fetching metrics data to match new API contract
- Renames `@/types/stats` -> `@/types/metrics`
- Adds data validation for metrics API response

### CODE REVIEW

File: frontend/src/components/ActivityDashboard/ActivityUsageSection/RecentInstalls.tsx [hunk 1]
```diff
@@ -5,11 +5,11 @@ import { usePluginMetrics } from '@/hooks';
 export function RecentInstalls() {
   const { plugin } = usePluginState();
   const { data: metrics, isLoading } = usePluginMetrics(plugin?.name);
-  const stats = metrics?.activity.stats;
+  const stats = metrics?.usage.stats;
 
   return (
     <RecentStats
-      count={stats?.installsInLast30Days}
```
Comment (manasaV3) [inline frontend/src/components/ActivityDashboard/ActivityUsageSection/RecentInstalls.tsx:12]: This should `installs_in_last_30_days`

File: frontend/src/components/ActivityDashboard/ActivityUsageSection/TotalInstalls.tsx [hunk 2]
```diff
@@ -5,11 +5,11 @@ import { usePluginMetrics } from '@/hooks';
 export function TotalInstalls() {
   const { plugin } = usePluginState();
   const { data: metrics, isLoading } = usePluginMetrics(plugin?.name);
-  const stats = metrics?.activity.stats;
+  const stats = metrics?.usage.stats;
 
   return (
     <TotalStats
-      count={stats?.totalInstalls}
```
Comment (manasaV3) [inline frontend/src/components/ActivityDashboard/ActivityUsageSection/TotalInstalls.tsx:12]: This should be `.total_installs`

**Review by manasaV3 — CHANGES_REQUESTED** [review]

File: frontend/src/components/ActivityDashboard/ActivityDashboard.tsx [hunk 3]
```diff
@@ -29,7 +29,7 @@ export function ActivityDashboard() {
       return true;
     }
 
-    const { timeline, stats } = metrics.activity;
+    const { timeline, stats } = metrics.usage;
```
Comment (klai95) [inline frontend/src/components/ActivityDashboard/ActivityDashboard.tsx:32]: A related question: where would we include `metrics.maintenance` that the backend PRs have?

File: frontend/src/utils/validate/validateMetricsData.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+const maintenanceSchema = z.object({
+  stats: z.object({
+    latest_commit_timestamp: z.string().nullable(),
+  }),
+});
+
+const usageSchema = z.object({
+  stats: z.object({
+    installs_in_last_30_days: z.number(),
```
Comment (klai95) [inline frontend/src/utils/validate/validateMetricsData.ts:15]: More of a heads-up than a review: if I make any changes to the name of the fields in my backend PRs, I will let you know to ensure frontend and backend abide by the same API contract

**Review by manasaV3 — APPROVED** [review]

**Pull request merged**